src/test/safe/SimpleETHContributionVault.t.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.19;
+
+import "forge-std/Test.sol";
+import {MockERC20} from "src/test/utils/mocks/MockERC20.sol";
+import {SimpleETHContributionVault} from "src/safe-modules/SimpleETHContributionVault.sol";
+
+contract SimpleETHContributionVaultTest is Test {
+  error CannotRageQuit();
+  error Unauthorized(address user);
+
+  event Deposit(address to, uint256 amount);
+  event DepositValidator(
+    bytes[] pubkeys, bytes[] withdrawal_credentials, bytes[] signatures, bytes32[] deposit_data_roots
+  );
+  event RageQuit(address to, uint256 amount);
+  event RescueFunds(uint256 amount);
+
+  address constant ETH_DEPOSIT_CONTRACT = 0x00000000219ab540356cBB839Cbe05303d7705Fa;
+  uint256 internal constant ETH_STAKE = 32 ether;
+
+  SimpleETHContributionVault contributionVault;
+
+  address safe;
+  address user1;
+  address user2;
+  address user3;
+  MockERC20 mERC20;
+
+  function setUp() public {
+    uint256 mainnetBlock = 17_421_005;
+    vm.createSelectFork(getChain("mainnet").rpcUrl, mainnetBlock);
+
+    safe = makeAddr("safe");
+    contributionVault = new SimpleETHContributionVault(
+            safe,
+            ETH_DEPOSIT_CONTRACT
+        );
+
+    mERC20 = new MockERC20("Test Token", "TOK", 18);
+    mERC20.mint(type(uint256).max);
+  }
+
+  function test_deposit() external {
+    vm.deal(user1, ETH_STAKE);
+
+    vm.expectEmit(false, false, false, true);
+    emit Deposit(user1, ETH_STAKE);
```

Comment:
Just for my understanding, what are these two lines doing exactly? We're expecting the emission of an event that has only one indexed event (the event name?), and then we actually emit such an event on line 48? Won't sending ether to the contract on line 51 cause a second Deposit event to be emitted? that this line on 48 makes redundant/means 47 will always pass? 

You do it on other functions though so I have a hunch I don't understand the test framework appropriately though.

Reply:
So expectEmit definition is below. 
```
expectEmit(
    bool checkTopic1,
    bool checkTopic2,
    bool checkTopic3,
    bool checkData
) external;
```
The three false are because the Deposit event doesn't have any indexed parameters. The final true param means to check non indexed parameters too